plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -385,30 +383,62 @@ public void updateTableStatistics(HiveIdentity identity, String databaseName, St
     }
 
     @Override
-    public void updatePartitionStatistics(HiveIdentity identity, Table table, String partitionName, Function<PartitionStatistics, PartitionStatistics> update)
+    public void updatePartitionStatistics(HiveIdentity identity, Table table, Map<String, Function<PartitionStatistics, PartitionStatistics>> updates)
     {
-        List<String> partitionValues = toPartitionValues(partitionName);
-        Partition partition = getPartition(identity, table, partitionValues)
-                .orElseThrow(() -> new TrinoException(HIVE_PARTITION_DROPPED_DURING_QUERY, "Statistics result does not contain entry for partition: " + partitionName));
+        Iterables.partition(updates.entrySet(), BATCH_CREATE_PARTITION_MAX_PAGE_SIZE).forEach(partitionUpdates ->
+                updatePartitionStatisticsBatch(table, partitionUpdates.stream().collect(toImmutableMap(Entry::getKey, Entry::getValue))));
+    }
 
-        PartitionStatistics currentStatistics = getPartitionStatistics(ImmutableList.of(partition)).get(partition);
-        PartitionStatistics updatedStatistics = update.apply(currentStatistics);
+    private void updatePartitionStatisticsBatch(Table table, Map<String, Function<PartitionStatistics, PartitionStatistics>> updates)
+    {
+        ImmutableList.Builder<BatchUpdatePartitionRequestEntry> updateRequests = ImmutableList.builder();
+        ImmutableSet.Builder<GlueColumnStatisticsProvider.PartitionStatisticsUpdate> statisticsUpdates = ImmutableSet.builder();
+
+        Map<List<String>, String> partitionValuesToName = updates.keySet().stream()
+                .collect(toImmutableMap(HiveUtil::toPartitionValues, identity()));
+        List<PartitionValueList> partitionValues = partitionValuesToName.keySet().stream()
+                .map(partition -> new PartitionValueList().withValues(partition))
+                .collect(toImmutableList());
+
+        BatchGetPartitionRequest batchGetPartitionRequest = new BatchGetPartitionRequest()
+                .withCatalogId(catalogId)
+                .withDatabaseName(table.getDatabaseName())
+                .withTableName(table.getTableName())
+                .withPartitionsToGet(partitionValues);
+        BatchGetPartitionResult batchGetPartitionResult = stats.getGetPartition().call(() -> glueClient.batchGetPartition(batchGetPartitionRequest));
+
+        GluePartitionConverter partitionConverter = new GluePartitionConverter(table);
+        Set<Partition> partitions = batchGetPartitionResult.getPartitions().stream().map(partitionConverter).collect(toImmutableSet());
+        Map<Partition, Map<String, HiveColumnStatistics>> statisticsPerPartition = columnStatisticsProvider.getPartitionColumnStatistics(partitions);
+
+        statisticsPerPartition.forEach((partition, columnStatistics) -> {
```

Comment:
Just to double check. If there are no statistics present in Glue for a partition we will still get an entry in map returned by `columnStatisticsProvider.getPartitionColumnStatistics`

Reply:
Yes